### PR TITLE
AIRFLOW-51 && AIRFLOW-71 docker_operator improvements

### PR DESCRIPTION
**Pull request by asnir**
AIRFLOW-51 Rename param Docker cpus -> cpu_shares
AIRFLOW-71 docker_operator - pulling from private repositories

https://github.com/apache/incubator-airflow/pull/1476
It can be merged at version 1.7

### CODE REVIEW

**Comment by asnir** [comment]
@mistercrunch, please approve it.

**Comment by bolkedebruin** [comment]
@asnir Thanks for the contribution! Before we can merge please provide a description in the PR, and squash your commits with proper messages as per guidelines. Thanks.

File: airflow/operators/docker_operator.py [hunk 1]
```diff
@@ -110,21 +118,28 @@ def __init__(
         self.volumes = volumes or []
         self.xcom_push = xcom_push
         self.xcom_all = xcom_all
+        self.dockercfg_path = dockercfg_path
+        self.registry_username = registry_username
 
         self.cli = None
         self.container = None
 
+        warnings.warn(
```
Comment (criccomini) [inline airflow/operators/docker_operator.py:127]: Shouldn't this only trigger if `cpus` is set?

File: airflow/operators/docker_operator.py [hunk 2]
```diff
@@ -71,7 +76,8 @@ def __init__(
             image,
             api_version=None,
             command=None,
-            cpus=1.0,
+            cpus=None,
```
Comment (criccomini) [inline airflow/operators/docker_operator.py:79]: I think @mistercrunch was asking you to make this change in a backwards compatible way in your other PR. This looks to me like you've just left the parameter dangling. It doesn't appear to be used anymore. This isn't backwards compatible.

**Pull request closed without merging**